src/mjolnir/ingest_transit.cc

```diff
@@ -524,6 +534,17 @@ bool write_stop_pairs(Transit& tile,
         }
       }
     }
+
+    tile_itr = tile_tripIds.erase(tile_itr);
+
+#if GOOGLE_PROTOBUF_VERSION >= 3001000
+    auto size = tile.ByteSizeLong();
+#else
+    auto size = tile.ByteSize();
+#endif
+    if (size > 134217728) {
+      return dangles;
+    }
```

Comment:
alright, took a stab at limiting PBF size (for ref PBF can't have a byte size of > 2 GB, apparently because their internal size check is based on `int`, not `size_t`, wtf!!).

we'd need to do smth like this IMO. here in `write_stop_pairs` we add most of the PBF objects and is likely the place where it goes wrong. also the old code was only checking the stop pairs for PBF size. but the old code was basically paging through transit.land's API response of `get_stop_pairs` endpoint and never really had the full picture of the entire tile available. however, we do have all GTFS objects this entire tile contains all at once. 

we can remove the the trip from the `transit_info` here and check in the calling function if there's still trips left to process.